src/Tomcat/TomcatServer.ts

```diff
@@ -5,21 +5,16 @@ import * as vscode from "vscode";
 import { ServerState } from "../Constants";
 import { Utility } from "../Utility";
 
-export class TomcatServer {
-    public needRestart: boolean = false;
-    private _name: string;
-    private _tomcatPath: string;
-    private _extensionPath: string;
-    private _debugPort: number;
-    private _debugWorkspace: vscode.WorkspaceFolder;
-    private _isDebugging: boolean = false;
-    private _state: ServerState = ServerState.IdleServer;
-
-    constructor(name: string, tomcatPath: string, extensionPath: string) {
-        this._name = name.trim();
-        this._tomcatPath = tomcatPath.trim();
-        this._extensionPath = extensionPath.trim();
-    }
+export class TomcatServer implements vscode.QuickPickItem {
+    // tslint:disable:no-parameter-properties no-unnecessary-initializer *
+    constructor(private _name: string, private _tomcatPath: string, private _extensionPath: string,
+                private _state: ServerState = ServerState.IdleServer, private _started: boolean = false,
+                private _isDebugging: boolean = false, private _debugPort: number = undefined,
+                private _debugWorkspace: vscode.WorkspaceFolder = undefined, public needRestart: boolean = false,
```

Comment:
keep the first three arguments only private _name: string, private _tomcatPath: string, private _extensionPath: string